src/python/pants/backend/jvm/tasks/scalafmt.py

```diff
@@ -51,15 +58,46 @@ def invoke_tool(self, _, target_sources):
     # If no config file is specified use default scalafmt config.
     config_file = self.get_options().configuration
     args = list(self.additional_args)
-    args.extend(['--files', ','.join(source for _, source in target_sources)])
-    if config_file != None:
+    if config_file is not None:
       args.extend(['--config', config_file])
+    if self.get_options().output_dir:
+      args.append('--stdout')
+
+    result = 0
+    created_dirs = set()
+
+    for _, source in target_sources:
+      res, workunit = self.runjava(classpath=self.tool_classpath('scalafmt'),
```

Comment:
So previously we executed once for all files, and now we do so once per file? That seems like a major negative performance impact in all cases, just to support this one use-case. 

Is it not possible to separate out the stdout output of running on multiple files? Is there no delimiter or something?